Hana/src/Platform/OpenGL/OpenGLVertexArray.cpp

```diff
@@ -0,0 +1,81 @@
+#include "hnpch.h"
+#include "OpenGLVertexArray.h"
+
+#include <glad/glad.h>
+
+namespace Hana
+{
+	static GLenum ShaderDataTypeToOpenGLBaseType(ShaderDataType type)
+	{
+		switch (type)
+		{
+			case Hana::ShaderDataType::Float:	return GL_FLOAT;
+			case Hana::ShaderDataType::Float2:	return GL_FLOAT;
+			case Hana::ShaderDataType::Float3:	return GL_FLOAT;
+			case Hana::ShaderDataType::Float4:	return GL_FLOAT;
+			case Hana::ShaderDataType::Mat3:	return GL_FLOAT;
+			case Hana::ShaderDataType::Mat4:	return GL_FLOAT;
+			case Hana::ShaderDataType::Int:		return GL_INT;
+			case Hana::ShaderDataType::Int2:	return GL_INT;
+			case Hana::ShaderDataType::Int3:	return GL_INT;
+			case Hana::ShaderDataType::Int4:	return GL_INT;
+			case Hana::ShaderDataType::Bool:	return GL_BOOL;
+		}
+
+		HN_CORE_ASSERT(false, "Unkonwn ShaderDataType!");
+		return 0;
+	}
+
+	OpenGLVertexArray::OpenGLVertexArray()
+	{
+		glCreateVertexArrays(1, &m_RendererID);
+	}
+
+	OpenGLVertexArray::~OpenGLVertexArray()
+	{
+		glDeleteVertexArrays(1, &m_RendererID);
+	}
+
+	void OpenGLVertexArray::Bind() const
+	{
+		glBindVertexArray(m_RendererID);
+	}
+
+	void OpenGLVertexArray::Unbind() const
+	{
+		glBindVertexArray(0);
+	}
+
+	void OpenGLVertexArray::AddVertexBuffer(const std::shared_ptr<VertexBuffer>& vertexBuffer)
+	{
+		HN_CORE_ASSERT(vertexBuffer->GetLayout().GetElements().size(), "Vertex Buffer has no layout!");
+
+		glBindVertexArray(m_RendererID);
+		vertexBuffer->Bind();
+
+		uint32_t index = 0;
+		const BufferLayout& layout = vertexBuffer->GetLayout();
+		for (const BufferElement& element : layout)
+		{
+			glEnableVertexAttribArray(index);
+			glVertexAttribPointer(index,
+				element.GetComponentCount(),
+				ShaderDataTypeToOpenGLBaseType(element.Type),
+				element.Normalized ? GL_TRUE : GL_FALSE,
+				layout.GetStride(),
+				(const void*)element.Offset);
+			index++;
+		}
+
+		m_VertexBuffers.push_back(vertexBuffer);
```

Comment:
The reason we want to support multiple vertex buffers and single index buffer in a vertex array is, each vertex array may contain different types of information ( position, texture coordinate, color, etc.) but with the same index layout.